### ISSUE TITLE
Canary Release

### ISSUE THREAD

**Issue by daviddykeuk** [turbosonic/event-hub-sidecar#12]
## tl:dr 
When deploying a new version of an event listening service I want it to slowly listen for events and check that they are being handled correctly before accepting them a full speed and pulling down the old versions of the service.

## Detail
The aim of this issue is to define a pattern whereby you can do canary releases of event listeners that will only deal with a very small amount of event traffic to test that logic/code works correctly before allowing a normal rate of events to be streamed and pulling down existing pods.

e.g. a new release of a worker pod would have a sidecar that only accepts one event at a time and has a pause (definable by environment) of perhaps 10 seconds between events, that pause would continue as time goes on (definable by environment) for two minutes until at which point the pod will switch to accepting events at full speed. At that point the pod would be declared ready and the next pod can be started (and an old one removed).

If, during the canary period, any events fail, the sidecar would exit with a failure and the deployment would not complete, this should be handled by the orchestrator to keep the existing version of the service. The event would be placed back on the queue to be handled by an existing service

## Changes required
At the moment events are dealt with by goroutines, increasing the throughput by processing them concurrently on multiple threads. To have this work correctly we would need to adapt this to stop events being processes concurrently and have a sleep mechanism at the end of each event process. This sleep would need to dynamically lower and when down to zero the concurrency can start again.

Environment variables would determine the rate of events on startup and the time to normal flow, i.e. to have the service accept one event every ten seconds for two minutes the setup would be...
* CANARY_EVENTS_PER_MINUTE=6
* CANARY_TIME=120
